cx/spark/server.js

```diff
@@ -0,0 +1,208 @@
+const request = require('request');
+const app = express();
+
+app.use(express.json());
+
+// Uncomment and insert your values here
+// const sparkAccessToken = "1234567898-ABCdfghTtaD8dfghdfgh-45sdf65467M";
+// const targetUrl = 'https://example.com';
+// const projectId = 'my-project';
+// const locationId = 'global';
+// const agentId = 'my-agent';
+// const languageCode = 'en'
+
+// Imports the Google Cloud Some API library
+const {SessionsClient} = require('@google-cloud/dialogflow-cx');
+/**
+ * Example for regional endpoint:
+ *   const locationId = 'us-central1'
+ *   const client = new SessionsClient({apiEndpoint: 'us-central1-dialogflow.googleapis.com'})
+ */
+const client = new SessionsClient({apiEndpoint: locationId + '-dialogflow.googleapis.com'});
+
+// Upon start a webhook is registered with spark
+// Upon closure the webhook is removed from spark
+
+const listener = app.listen(process.env.PORT, async function() {
+  await init();
+  console.log('Your Spark integration server is listening on port '
+      + listener.address().port);
+});
+
+app.post('/', async function(req, res) {
+  const message = await retrieveMessage(req.body.data.id);
+  const dialogflowResponse = await detectIntentText(message, req.body.data.personId);
+  const sparkMessage = detectIntentToSparkMessage(dialogflowResponse);
+  sendMessage(sparkMessage.text, message.email);
+});
+
+process.on('SIGTERM', () => {
+  listener.close(async ()=>{
+    console.log('Closing http server.');
+    await deleteWebhooks();
+    process.exit(0);
+  });
+});
+
+async function init(){
+  await deleteWebhooks();
+  registerWebhook();
+}
+
+// Converts Spark message to a detectIntent request. 
+function sparkToDetectIntent(message, sessionPath){
+     const request = {
+     session: sessionPath,
+     queryInput: {
+         text: {
+             text: message.text,
+            },
+            languageCode,
+        },
+    };
+
+  return request;
+}
+
+// Converts detectIntent response to a Spark text message. 
+function detectIntentToSparkMessage(response){
+    agentResponse = '';
+    
+    for (const message of response.queryResult.responseMessages) {
+        if (message.text) {
+            agentResponse += `${message.text.text}\n`;
+        };
+    };
+  
+  if(agentResponse.length != ''){
+    const request = {
+      text: agentResponse
+     };
+    return request;
+    };
+};
+
+/**
+ * This function calls Dialogflow CX API to retrieve the response
+ * https://cloud.google.com/dialogflow/cx/docs/quick/api
+ */
+async function detectIntentText(message, personId) {
+  const sessionId = personId;
+  const sessionPath = client.projectLocationAgentSessionPath(
+      projectId,
+      locationId,
+      agentId,
+      sessionId
+  );
+  console.info(sessionPath);
+
+  request = sparkToDetectIntent(message, sessionPath);
+  const response = await client.detectIntent(request);
+
+  return response;
+}
+
+function sendMessage(text, personEmail) {
```

Comment:
Would it be more flexible to accept the Spark message here? (E.g., if we would like to support sending card/button message later?)